index.php

```diff
@@ -231,8 +231,210 @@ function convertTo24HourFormat($time) {
+            "Year" => 2000
+        )
+    );
+
+    echo "<h2>Search Results for '$search_query'</h2>";
+    echo "<ul>";
+    $found = false;
+    foreach ($Books as $Books) {
+        if (stripos($Books["Title"], $search_query) !== false || stripos($Books["Author"], $search_query) !== false) {
+            echo "<li>";
+            echo "Title: " . $Books["Title"] . "<br>";
+            echo "Author: " . $Books["Author"] . "<br>";
+            echo "Year: " . $Books["Year"] . "<br>";
+            echo "</li>";
+            $found = true;
+        }
+    }
+    if (!$found) {
+        echo "<li>No books found for the search term '$search_query'.</li>";
+    }
+    echo "</ul>";
+}
+
+//=======================================================================================
+//Q#10:
+//display errors in php
+//Implement error handling to display errors when the form is submitted with missing or incorrect data.
+//Ensure that user-friendly error messages are displayed.
+
+if ($_SERVER["REQUEST_METHOD"] == "GET" && isset($_GET['search_query'])) {
+    $search_query = trim($_GET['search_query']);
+    $error_message = "";
+
+    // Validate input
+    if (empty($search_query)) {
+        $error_message = "Search query cannot be empty. Please enter a title or author to search.";
+    } elseif (!preg_match("/^[a-zA-Z0-9\s]+$/", $search_query)) {
+        $error_message = "Search query contains invalid characters. Please use only letters, numbers, and spaces.";
+    }
 
+    // Display error message if any
+    if (!empty($error_message)) {
+        echo "<p style='color: red;'>$error_message</p>";
+    } else {
+        $Books = array(
+            array(
+                "Title" => "PHP",
+                "Author" => "W Jason Gilmore",
+                "Year" => 2003
+            ),
+            array(
+                "Title" => "Javascript",
+                "Author" => "David Herman",
+                "Year" => 2000
+            )
+        );
+
+        echo "<h2>Search Results for '$search_query'</h2>";
+        echo "<ul>";
+        $found = false;
+        foreach ($Books as $book) {
+            if (stripos($book["Title"], $search_query) !== false || stripos($book["Author"], $search_query) !== false) {
+                echo "<li>";
+                echo "Title: " . $book["Title"] . "<br>";
+                echo "Author: " . $book["Author"] . "<br>";
+                echo "Year: " . $book["Year"] . "<br>";
+                echo "</li>";
+                $found = true;
+            }
+        }
+        if (!$found) {
+            echo "<li>No books found for the search term '$search_query'.</li>";
+        }
+        echo "</ul>";
+    }
+}
 
 ?>
 
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Add New Book</title>
+</head>
+<body>
+    <h2>Add a New Book</h2>
+    <form action="add_book.php" method="post">
```

Comment:
sugggest to solve error if this error is occur

Error: Form submission method not allowed.
The error "Form submission method not allowed" occurs because the form processing logic may be running twice: once for the form submission and once for a separate part of the script that is not meant to handle form submissions. This happens if you have both POST and GET form handling logic in the same script without proper checks.

To handle multiple forms with the same method in a single script, we need to ensure that our script can correctly distinguish between the forms and handle each form's submission accordingly.

Let's combine both forms in a single script and use hidden inputs to identify which form is submitted.